neptunelib/utils.py

```diff
@@ -31,3 +37,47 @@ def as_list(value):
         return value
     else:
         return [value]
+
+
+def align_channels_on_x(dataframe):
+    x_max = _get_max_x(dataframe)
+    joined_x = pd.DataFrame({'x': list(range(x_max + 1))})
```

Comment:
This line will blow up user's RAM if `x` is a timestamp. A simple fix here would be getting an `x_min` as well, which fixes the problem for timestamps. However, this will still blow up on `x = [1, 2, 3, 10**10]`, which is a valid x sequence.

Reply:
You are absolutely right.
Let me think how to go about it.